src/sorunlib/wiregrid.py

```diff
@@ -305,3 +411,80 @@ def calibrate(continuous=False, elevation_check=True, boresight_check=True,
     finally:
         # Stop SMuRF streams
         run.smurf.stream('off')
+
+
+def time_constant(initial_hwp_direction, stepwise_first=True, stepwise_last=True, stepwise_mid=False, repeat=1):
+    """
+    Run a wiregrid time constant measurement.
+
+    Args:
+        initial_hwp_direction (str): Initial HWP direction, 'forward' or 'backward'.
+        stepwise_first (bool): Do stepwise rotation or not before the first HWP speed change. Default is True.
+        stepwise_last (bool): Do stepwise rotation or not after the last HWP speed change. Default is True.
+        stepwise_mid (bool): Do stepwise rotation between each HWP speed change. Default is False.
+        repeat (int): Number of repeats. Default is 1.
+            If this is odd, the HWP direction will be changed to the opposite of the initial direction.
+            If this is even, the HWP direction will be the same as the initial direction.
+    """
+
+    # Check the initial HWP direction
+    if initial_hwp_direction not in ['forward', 'backward']:
+        raise RuntimeError("Initial HWP direction should be either 'forward' or 'backward'. Aborting...")
+    current_hwp_direction = initial_hwp_direction
+
+    # Check the repeat
+    if repeat < 1 or not isinstance(repeat, int):
+        raise RuntimeError("The repeat should be int and larger than 0. Aborting...")
+
+    _check_agents_online()
+    _check_motor_on()
+    _check_telescope_position(elevation_check=True, boresight_check=False)
+
+    if _check_zenith:
+        el_tag = ', wg_el90'
+    else:
+        el_tag = ''
+
+    # Rotate for reference before insertion
+    rotate(continuous=True, duration=10)
+
+    # Bias step (the wire grid is off the window)
+    bs_tag = f'biasstep, wg_time_constant, wg_ejected, hwp_2hz_{current_hwp_direction}' + el_tag
+    run.smurf.bias_step(tag=bs_tag, concurrent=True)
+
+    # Insert the wiregrid
+    insert()
```

Comment:
Could you add detector streams during the insert() and eject(), and add sleep (5 sec?) before and after the insert()/eject()?
Sleep times have two purposes. One is to see the detector response with/without the wiregrid right before and after the insertion/ejection. The second is to wait for the wiregrid to be stable before the bias-step measurement.

Reply:
I added detector streams and sleep. In these streams, I used new tags, `wg_inserting` and `wg_ejecting`. The commit for these modification is here: f1e1157